lib/url-pattern/url-match-pattern-list.js

```diff
@@ -0,0 +1,95 @@
+var _ = require('../util').lodash,
```

Comment:
I added this class, so that we can have different approaches to matching a url with a pattern, and a list of patterns.

In case we add [cache mechanisms over internal regex matches](https://github.com/postmanlabs/postman-collection/blob/3706653574b239add3aaed128e8e9c52cfbe2b86/lib/collection/proxy-config.js#L148), this class might be useful over a static method.